src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -581,28 +574,27 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
                     "fn@",
                     "module@",
                     "method@",
-                ]
-                .iter()
-                .find(|p| link.starts_with(**p))
-                {
-                    kind = Some(ValueNS);
-                    link.trim_start_matches(prefix)
-                } else if link.ends_with("()") {
-                    kind = Some(ValueNS);
-                    link.trim_end_matches("()")
-                } else if link.starts_with("macro@") {
-                    kind = Some(MacroNS);
-                    link.trim_start_matches("macro@")
-                } else if link.starts_with("derive@") {
-                    kind = Some(MacroNS);
-                    link.trim_start_matches("derive@")
-                } else if link.ends_with('!') {
-                    kind = Some(MacroNS);
-                    link.trim_end_matches('!')
```

Comment:
re https://github.com/rust-lang/rust/pull/74048/files#r449882381

Quoting [intra doc link RFC](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#path-ambiguities):

> - In links to macros,
>  the link label can end with a `!`,
>  e.g., `Look at the [FOO!] macro`. You can alternatively use a `macro@` prefix,
>  e.g. `[macro@foo]`

I don't know if multiple `!` is allowed in that RFC. cc @Manishearth

Reply:
Only a single one